main.c

```diff
@@ -1657,6 +1665,53 @@ static struct PyModuleDef gmp_module = {
 };
 
 
+#define max_alloc 8192
+static void* mem_arr[max_alloc];
+size_t idx = 0;
+
+
+static void *
+my_allocate_function(size_t size)
+{
+    if (idx > max_alloc) {
+        goto err;
+    }
+
+    void *ret = malloc(size);
+
+    if (ret == 0) {
```

Comment:
```suggestion
    if (ret == NULL) {
```
(Even though they are the same, I still recommend using NULL.)